src/main/java/com/google/api/gax/bundling/ThresholdBundler.java

```diff
@@ -49,60 +49,85 @@
  */
 public final class ThresholdBundler<E> {
 
-  private ImmutableList<BundlingThreshold<E>> thresholdPrototypes;
+  private class ReleaseResourcesFunction implements Function<Object, Void> {
```

Comment:
No, this shouldn't be Function<ApiFuture<?>, Void>, it should be Function<?, Void>. We can't do this directly, but we can parametrize `ReleaseResourcesFunction`.